app/api/client_resources.py

```diff
@@ -266,10 +266,14 @@ def resource_create(
                 "remoteNetworkId": resource.remote_network_id,
                 "securityPolicyId": resource.security_policy_id,
                 "protocols": resource.protocols.model_dump(by_alias=True),
-                "tags": [
-                    {"key": key, "value": value}
-                    for key, value in k8s_metadata.labels.items()
-                ],
+                "tags": (
+                    [
+                        {"key": key, "value": value}
+                        for key, value in k8s_metadata.labels.items()
+                    ]
+                    if resource.sync_labels
+                    else []
+                ),
```

Comment:
I am missing a test here, we should probably mock `execute_mutation` then assert it is called with empty `list`. I will add it tomorrow